spec/models/post_spec.rb

```diff
@@ -1811,4 +1811,28 @@ def updates_topic_updated_at
       expect(post.cannot_permanently_delete_reason(Fabricate(:admin))).to eq(nil)
     end
   end
+
+  describe "#canonical_url" do
+    it 'is able to determine correct canonical urls' do
+
+      # ugly, but no interface to set this and we don't want to create
+      # 100 posts to test this thing
+      TopicView.stubs(:chunk_size).returns(2)
```

Comment:
Perhaps it might be time to introduce an interface instead of carrying a small technical debt here?

Reply:
Agree, but I am counting 12 stubs now. Probably better do this in a followup PR.